loader/src/aarch64/loader.c

```diff
@@ -263,6 +267,12 @@ putc(uint8_t ch)
 {
     *((volatile uint32_t *)(0x00FF000030)) = ch;
 }
+#elif defined(BOARD_ultra96v2) || defined(BOARD_ultra96v2_hyp)
+static void
+putc(uint8_t ch)
+{
+    *((volatile uint32_t *)(0x00FF010030)) = ch;
```

Comment:
Should this wait for the transmit to be empty? Something like https://github.com/seL4/seL4/blob/07a8263a9d5ccaafca9f94be1d3539a0cd3f9127/src/drivers/serial/xuartps.c#L38. In fact I believe the current ZCU102 UART in the loader is wrong since I recall when I tested on hardware (QEMU was fine) that there was something weird with the serial output.

I should update the ZCU102 UART myself.

Reply:
Yes, it should.
When I started with the board it was not 1st priority. So, I fixed it now and thanks for the link.